tests/components/DirectAnswer.stories.tsx

```diff
@@ -12,14 +18,69 @@ const meta: ComponentMeta<typeof DirectAnswer> = {
 };
 export default meta;
 
-export const FieldValue = (args: DirectAnswerProps) => {
+const baseDirectAnswerResult = fieldValueDAState.result as FieldValueDirectAnswerType;
+
+function generateFieldValueDirectAnswer(
+  args: DirectAnswerProps,
+  fieldType?: FieldValueDirectAnswerType['fieldType'],
+  value?: unknown,
+): JSX.Element {
   return (
     <SearchHeadlessContext.Provider value={generateMockedHeadless({
-      directAnswer: fieldValueDAState
+      directAnswer: {
+        result: {
+          ...baseDirectAnswerResult,
+          ...(fieldType !== undefined && { fieldType }),
+          ...(value !== undefined && { value }),
+        } as FieldValueDirectAnswerType
```

Comment:
is there a way this could be refactored so casting isn't needed?

Reply:
I couldn't figure out a nice way to do it. It seems to be the TS error (without using the type cast `as`) may stem from the `RecursivePartial` type util we have which makes it think the result where should be of the address (error: `Type '"unknown"' is not assignable to type 'EnumOrLiteral<BuiltInFieldType.Address>'`). I can dig deeper if you like, or leave it for now and have another item to investigate?

Reply:
got it. I think it's fine to leave as is then. I was just curious